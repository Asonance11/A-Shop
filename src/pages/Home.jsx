import React from 'react';
import { Link } from 'react-router-dom';

const Home = () => {
	return (
		<main className="pt-20 bg-darkGrey h-screen">
			<h1 className="absolute text-7xl font-serif italic top-[30%] left-[50%] translate-x-[-50%] z-10 text-white">
				Catching Moments
			</h1>
			<figure className="bg-[url(https://images.pexels.com/photos/1421903/pexels-photo-1421903.jpeg)] h-[70vh] bg-center bg-cover bg-no-repeat relative">
				<figcaption>
					<a
						className="absolute left-2 bottom-4 p-2 underline text-blue-500 bg-white rounded"
						href="https://www.pexels.com/@eberhardgross/"
					>
						Photo by Eberhard Grossgasteiger
					</a>
				</figcaption>
			</figure>
			<div className="flex items-center justify-between mt-4 px-8">
				<p className="text-xl text-white font-normal">
					Shop for framed photos from all around the world.
				</p>
				<button
					type="button"
					className="py-2 px-4 bg-gradient-to-r from-forestGreen via-adventurousYellow to-oceanTurquoise rounded-md text-darkGrey"
				>
					<Link to="/shop">Shop Now</Link>
				</button>
			</div>
		</main>
	);
};

export default Home;
