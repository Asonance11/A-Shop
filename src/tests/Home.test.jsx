import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { describe, it } from 'vitest';
import Home from '../pages/Home';

describe('Home component', () => {
	it('renders properly', () => {
		const { container } = render(<Home />, { wrapper: MemoryRouter });
		expect(container).toMatchSnapshot();
	});
});
