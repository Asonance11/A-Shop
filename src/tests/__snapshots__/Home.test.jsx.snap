// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component > renders properly 1`] = `
<div>
  <main
    class="pt-20 bg-darkGrey h-screen"
  >
    <h1
      class="absolute text-7xl font-serif italic top-[30%] left-[50%] translate-x-[-50%] z-10 text-white"
    >
      Catching Moments
    </h1>
    <figure
      class="bg-[url(https://images.pexels.com/photos/1421903/pexels-photo-1421903.jpeg)] h-[70vh] bg-center bg-cover bg-no-repeat relative"
    >
      <figcaption>
        <a
          class="absolute left-2 bottom-4 p-2 underline text-blue-500 bg-white rounded"
          href="https://www.pexels.com/@eberhardgross/"
        >
          Photo by Eberhard Grossgasteiger
        </a>
      </figcaption>
    </figure>
    <div
      class="flex items-center justify-between mt-4 px-8"
    >
      <p
        class="text-xl text-white font-normal"
      >
        Shop for framed photos from all around the world.
      </p>
      <button
        class="py-2 px-4 bg-gradient-to-r from-forestGreen via-adventurousYellow to-oceanTurquoise rounded-md text-darkGrey"
        type="button"
      >
        <a
          href="/shop"
        >
          Shop Now
        </a>
      </button>
    </div>
  </main>
</div>
`;
